### ISSUE TITLE
 `System.Text.StringBuilder' could be found.

### ISSUE THREAD

**Issue by WindRunnerCackerChen** [Unity-Technologies/ml-agents#557]
**[Drag the ML-Agents folder from unity-environments/Assets to the Unity Editor Project window.](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Learning-Environment-Create-New.md)And then print this error message.My queastion is why I missing this assembly reference while I Drag the same folder into the Unity Editor Project window and where is it?**

_Assets/ML-Agents/Scripts/ExternalCommunicator.cs(259,24): error CS1061: Type `System.Text.StringBuilder' does not contain a definition for `Clear' and no extension method `Clear' of type `System.Text.StringBuilder' could be found. Are you missing an assembly reference?_

Thank you so much!

**Comment by MarcoMeter** [comment]
I assume you created a new project and tried to import all the ml-agents assets, right?
If so, make sure that you are running on .NET 4.6:

Edit -> Project Settings -> Player
Then in the inspector under "Other Settings" select the .NET 4.6 equivalent Scripting Runtime Version.

**Comment by WindRunnerCackerChen** [comment]
@MarcoMeter Yeah,you got it,but I found a way that open my new project  with double clicking one of the scene from the example folder to avoid this error,it will be fixed automatically by Unity Editor loading the .meta files. I will check my Runtime Version when I import all the ml-agents assets again,and I will reply to you as quickly as possible.

**Comment by gw1108** [comment]
Is your editor version 2018.1.0b6?
I assume that's important.

**Comment by gw1108** [comment]
I had a project that had this error, but I was using Unity 2017.3.1f1. Using Unity 2018.1.0b13 and no errors.

**Comment by WindRunnerCackerChen** [comment]
@gw1108 2017.3.1f1 is my version.

**Comment by entropia666** [comment]
I had exactly the same problem and thanks to MarcoMeter it was solved. Thanks! I had selected the .net 3.5 instead of 4.6.

**Comment by mmattar** [comment]
Hi @WindRunnerCackerChen - did the feedback on this thread resolve the issue? Let us know if you need any additional support.

**Comment by FortyFortyLabs** [comment]
had same problem but resolution from MarcoMeter seems to have fixed it , thanx.

**Comment by h1r03** [comment]
had the same problem and was able to fix it now thanks to MarcoMeter. (I used 2017.4.1f1 Personal just installed yesterday on OSX.) It might be nicer to mention this in this tutorial.

Also before noticing this issue, I originally stuck in [Select the Scripts/Brain](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Learning-Environment-Create-New.md) since I could not find Scripts/Brain component. But this is also fixed and I can see Scripts/Brain component. Thank you!

<img src="https://user-images.githubusercontent.com/8764683/39097806-7e3d998c-4616-11e8-97be-6c7686b7edb9.png" height="200">

**Comment by WindRunnerCackerChen** [comment]
@mmattar definitely done.thanks
